extensions/amp-story/1.0/amp-story-open-page-attachment.js

```diff
@@ -108,6 +127,75 @@ const renderDefaultPageAttachmentUI = (pageEl, attachmentEl) => {
   return openAttachmentEl;
 };
 
+/**
+ * Renders inline page attachment UI.
+ * @param {!Window} win
+ * @param {!Element} pageEl
+ * @param {!Element} attachmentEl
+ * @return {!Element}
+ */
+ const renderOutlinkPageAttachmentUI = (win, pageEl, attachmentEl, attachmentHref) => {
+  const openAttachmentEl = buildOpenOutlinkAttachmentElement(pageEl);
+
+  // Copy href to the element so it can be previewed on hover and long press.
+  if (attachmentHref) {
+    openAttachmentEl.setAttribute('href', attachmentHref);
+  }
+  
+  openAttachmentEl.addEventListener('click', () =>
+    openAttachmentEl.openAttachment()
+  );
+
+  // Setting theme  
+  const theme = attachmentEl.getAttribute('theme');
+  if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
+    openAttachmentEl.setAttribute('theme', AttachmentTheme.DARK);
+  }
+
+  // Appending text & aria-label.
+  const openLabelAttr = attachmentEl.getAttribute('data-cta-text');
+  const openLabel =
+    (openLabelAttr && openLabelAttr.trim()) ||
+    getLocalizationService(pageEl).getLocalizedString(
+      LocalizedStringId.AMP_STORY_PAGE_ATTACHMENT_OPEN_LABEL
+    );
+
+  const ctaLabelEl = openAttachmentEl.querySelector(
+    '.i-amphtml-story-outlink-page-attachment-label'
+  );
+  ctaLabelEl.innerHTML = openLabel;
+  openAttachmentEl.setAttribute('aria-label', openLabel);
+
+  // Adding image.
+  const openImgAttr = attachmentEl.getAttribute('cta-image');
+  const chipEl = openAttachmentEl.querySelector(
+    '.i-amphtml-story-outlink-page-attachment-outlink-chip'
+  );
+
+  if (!attachmentEl.hasAttribute('cta-image') || openImgAttr !== 'none') {
+    const ctaImgEl = win.document.createElement('div');
+    ctaImgEl.classList.add('i-amphtml-story-outlink-page-attachment-img');
+    if (!attachmentEl.hasAttribute('cta-image')) {
+      ctaImgEl.classList.add(
+        'i-amphtml-story-outlink-page-attachment-default-img'
+      );
+    } else {setImportantStyles(ctaImgEl, {
+        'background-image': 'url(' + openImgAttr + ')',
+      });
+    }
+    chipEl.prepend(ctaImgEl);
+    setImportantStyles(ctaLabelEl, {
+      'padding-left': '6px',
+    });
+  } else {
+    setImportantStyles(chipEl, {
+      'padding-left': '14px',
+    });
```

Comment:
Can't this be rewritten with CSS?

Reply:
The problem is that it is conditional styling. According to the figma mocks:
- If there is no image, padding should be 14px on both sides of text
- If there is an image, there should be 6px between the text and the image and 6px between the image and the edge

The only way to do this with CSS would be to create new classes and then adding them conditionally to the elements. I don't personally think that's cleaner though?

Reply:
Can't you do `[cta-img] .chip-class`? Agree adding a new class is not necessary

Reply:
Ok I thought about it and you're right, it's easier to debug if it's in the CSS.

I tried to do it with the selector you recommended, but they don't actually share a parent element bc they're in the shadow DOM. It was cleaner to just append some tiny classes.